app/src/test/java/com/iic/shopingo/services/UserConnectorTest.java

```diff
@@ -0,0 +1,144 @@
+package com.iic.shopingo.services;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import bolts.Task;
+import com.facebook.FacebookRequestError;
+import com.facebook.Request;
+import com.facebook.RequestAsyncTask;
+import com.facebook.Response;
+import com.facebook.Session;
+import com.facebook.model.GraphUser;
+import com.iic.shopingo.dal.models.User;
+import mockit.Delegate;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Mocked;
+import mockit.NonStrictExpectations;
+import mockit.Verifications;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test for UserConnector
+ * Created by ifeins on 3/4/15.
+ */
+public class UserConnectorTest {
+
+  private UserConnector subject;
+
+  @Injectable
+  private SharedPreferences sharedPreferences;
+
+  @Injectable
+  private Session fbSession;
+
+  @Mocked
+  private Request anyFbRequest;
+
+  @Injectable
+  private Response fbResponse;
+
+  private Request.GraphUserCallback fbCallback;
+
+  @Before
+  public void beforeEach() {
+    subject = new UserConnector(sharedPreferences);
+    SharedUserConnector.setInstance(subject);
+
+    // we want this behavior to occur in all the tests that exercise the Request.newMeRequest static method.
+    // Basically what we want it do is to return a mocked instance of Request and store the callback in
+    // a member variable, so we can later call methods on it.
+    new NonStrictExpectations() {{
+      Request.newMeRequest(fbSession, (Request.GraphUserCallback) any);
+      result = new Delegate<Request>() {
+        Request delegate(Session session, Request.GraphUserCallback callback) {
+          fbCallback = callback;
+          return anyFbRequest;
+        }
+      };
+    }};
+  }
+
+  @After
+  public void afterEach() {
+    SharedUserConnector.setInstance(null);
+  }
+
+  @Test
+  public void testConnectWithFacebookWhenLoginFails() {
+    new Expectations() {{
+      fbResponse.getError();
+      result = new FacebookRequestError(0, "no_internet", "No internet");
+
+      anyFbRequest.executeAsync();
+      result = new Delegate<RequestAsyncTask>() {
+        RequestAsyncTask delegate() {
+          fbCallback.onCompleted(null, fbResponse);
+          return null;
+        }
+      };
+    }};
+
+    Task<User> task = subject.connectWithFacebook(fbSession);
+    Assert.assertTrue("task should be faulted", task.isFaulted());
+  }
+
+  /**
+   * It should return a promise that resolves a new User, and set it as the current user.
+   */
+  @Test
+  public void testConnectWithFacebookWhenLoginSucceeds(@Injectable final GraphUser graphUser) {
+    new Expectations() {{
+      graphUser.getId(); result = "fb-id-1";
```

Comment:
What do you think about moving all the expected results to variables, instead of having to repeat them in the assertions?